### PR TITLE
Add LSP/DAP configuration section on external editors

### PR DESCRIPTION
**Pull request by rsubtil**
[DAP](https://microsoft.github.io/debug-adapter-protocol/) has been undocumented ever since [it's implementation](https://github.com/godotengine/godot/pull/50454) in 2021 (wow, time flies :upside_down_face:). Likewise, [LSP](https://microsoft.github.io/language-server-protocol/) doesn't seem to be documented either, so I took the opportunity to add it as well.

The purpose of this section is to not only document that these features exist and how to use them, but to give general pointers to existing official and community guides/configs to be able to use them. So far I've only included steps for editors Godot has official plugin support, but there are more guides for other editors, such as [neovim](https://github.com/mfussenegger/nvim-dap/wiki/Debug-Adapter-installation#godot-gdscript).

### CODE REVIEW

**Comment by AnderRasoVazquez** [comment]
In my setup on Godot 4.1.2 I needed "debugServer" also:

```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "GDScript Godot",
            "type": "godot",
            "request": "launch",
            "project": "${workspaceFolder}",
            "port": 6005,
            "debugServer": 6006,
            "address": "127.0.0.1",
            "launch_game_instance": true,
            "launch_scene": false
        }
    ]
}
```

**Comment by rsubtil** [comment]
> In my setup on Godot 4.1.2 I needed "debugServer" also:

Yes, this is necessary in order to switch to DAP over the plugin's debug implementation, it's in this documentation proposal.

File: tutorials/editor/external_editor.rst [hunk 1]
```diff
@@ -68,3 +68,43 @@ We have official plugins for the following code editors:
 
 - `Visual Studio Code <https://github.com/godotengine/godot-vscode-plugin>`_
 - `Emacs <https://github.com/godotengine/emacs-gdscript-mode>`_
+
+LSP/DAP support
+---------------
+
+Godot supports the `Language Server Protocol <https://microsoft.github.io/language-server-protocol/>`_ (**LSP**) for code completion and the `Debug Adapter Protocol <https://microsoft.github.io/debug-adapter-protocol/>`_ (**DAP**) for debugging. You can check the `LSP client list <https://microsoft.github.io/language-server-protocol/implementors/tools/>`_ and `DAP client list <https://microsoft.github.io/debug-adapter-protocol/implementors/tools/>`_ to find if your editor supports them, and if yes, you should be able to take advantage of these features without the need of a custom plugin.
```
Comment (Calinou) [inline tutorials/editor/external_editor.rst:75]:
```suggestion
Godot supports the `Language Server Protocol <https://microsoft.github.io/language-server-protocol/>`_ (**LSP**) for code completion and the `Debug Adapter Protocol <https://microsoft.github.io/debug-adapter-protocol/>`_ (**DAP**) for debugging. You can check the `LSP client list <https://microsoft.github.io/language-server-protocol/implementors/tools/>`_ and `DAP client list <https://microsoft.github.io/debug-adapter-protocol/implementors/tools/>`_ to find if your editor supports them. If this is the case, you should be able to take advantage of these features without the need of a custom plugin.
```

File: tutorials/editor/external_editor.rst [hunk 2]
```diff
@@ -68,3 +68,43 @@ We have official plugins for the following code editors:
...
+
+To use these protocols, a Godot instance must be running on your current project. You should then configure your editor to communicate to the running adapter ports in Godot, which by default are ``6005`` for **LSP**, and ``6006`` for **DAP**. You can change these ports and other settings in the **Editor Settings**, under the **Network > Language Server** and **Network > Debug Adapter** sections respectively.
+
+Below are some configuration steps for specific editors:
+
+- Visual Studio Code
```
Comment (Calinou) [inline tutorials/editor/external_editor.rst:81]:
```suggestion
Visual Studio Code
^^^^^^^^^^^^^^^^^^
```

File: tutorials/editor/external_editor.rst [hunk 3]
```diff
@@ -68,3 +68,43 @@ We have official plugins for the following code editors:
...
+                "type": "godot",
+                "request": "launch",
+                "project": "${workspaceFolder}",
+                "port": 6007,
+                "debugServer": 6006,
+            }
+        ]
+    }
+
+- Emacs
```
Comment (Calinou) [inline tutorials/editor/external_editor.rst:107]:
```suggestion
Emacs
^^^^^
```

**Review by Calinou — APPROVED** [review]
Thanks!

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.1.

**Pull request merged**